File: Reverse_WordInPlace_String/Reverse.java
/*
    Author : Pakshal Shashikant Jain
    Date : 26/09/2023
    Program : Write a program which accept string from user and reverse each
              word in place.
              
              Input : "Marvellous Multi OS
              Output : "suollevraM itluM SO" 
*/

///////////////////////////////////////////////////////////////////////////////////

//Required Libraries
import java.lang.*;
import java.util.*;

///////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class StringRev
{
    //Characteristics of Class
    String str;
    char arr[];

    //Constructor of Class
    StringRev()
    {
        this.str = " ";
        this.arr = null;
    }

    //Behaviour of Class To Take Input from user
    void Accept()
    {
        //Creating object of Scanner Class To Take Input from user
        Scanner sobj = new Scanner(System.in);
        this.str = sobj.nextLine();

        this.arr = new char[this.str.length() + 1];

        this.arr = this.str.toCharArray();
    }

    //Behaviour of Class To Reverse Words of Entered Stroing
    void StrWrdRev()
    {
        int first = 0;
        int last = 0;
        int i = 0;
        int j = 0;
        char temp = ' ';

        //Logic To Reverse
        while(last != this.arr.length)
        {
            if(this.arr[last] == ' ')
            {
                while((this.arr[last] == ' ')&&(last != this.arr.length))
                {
                    last++;
                }
                last--;
            }   
            else
            {   
                first = last;
                while((this.arr[last] != ' ')&&(last != this.arr.length - 1))
                {
                    last++;
                }
                
                if(this.arr[last] == ' ')
                {
                    last--;
                }
                
                i = first;
                j = last;
                
                while(i <= j)
                {
                    temp = this.arr[i];
                    this.arr[i] = this.arr[j];
                    this.arr[j] = temp;

                    i++;
                    j--;
                }
            }
            last++;
        }
    }
}

//////////////////////////////////////////////////////////////////////////////////////////////

//Main Class
class Reverse
{
    //Main Function
    public static void main(String args[])
    {
        System.out.println("Jay Ganesh....");

        //Creating object of StringRev Class   
        StringRev sobj = new StringRev();

        //Call To Behaviours of Class
        sobj.Accept();
        sobj.StrWrdRev();

        System.out.print(sobj.arr);
    }
}

File: Reverse_ToggledCase_String/Reverse.go
/*
    Author : Pakshal Shashikant Jain 
    Date : 18/09/2023
    Program : Accept sing from user and reverse the contents of that string by
              toggling the case.

              Input : "aCBdef"
              Output : "FEDcbA" 
*/

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
package main
import (
	"fmt"
	"bufio"
	"os"
)

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function To Reverse String After Toggling Case
func StrRevTogX(str *[]byte) {
	var first int = 0;
	var last int = 0;
	var temp byte = ' ';

	//Logic To Reverse string
	for (last != len(*str)) {
		last++;
	}

	last--;

	for (first <= last) {

		if (((*str)[first] >= 'A')&&((*str)[first] <= 'Z')) {
			(*str)[first] = (*str)[first] + 32;
		}else if (((*str)[first] >= 'a')&&((*str)[first] <= 'z')) {
			(*str)[first] = (*str)[first] - 32;
		}

		if (last != first) {
			if (((*str)[last] >= 'A')&&((*str)[last] <= 'Z')) {
				(*str)[last] = (*str)[last] + 32;
			} else if (((*str)[last] >= 'a')&&((*str)[last] <= 'z')) {
				(*str)[last] = (*str)[last] - 32;
			}				
		}

		temp = (*str)[first];
		(*str)[first] = (*str)[last];
		(*str)[last] = temp;
		first++;
		last--;
	}
}//End of Function

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
func main() {
	var arr []byte;
	var str string;
	var err error = nil;
	
	//Creating object of bufio class To Take Input from user
	var bobj =  bufio.NewReader(os.Stdin);

	fmt.Println("Jay Ganesh....");

	//Taking Input from user
	fmt.Println("Enter One String");

	str,err = bobj.ReadString('\n');
	if (err != nil) {
		return;
	}
	
	//Creating character Array of Entered String
	arr = make([]byte,len(str));
	arr = []byte(str);

	//Call To StrRevTogX function
	StrRevTogX(&arr);

	//Printing Reversed string
	fmt.Print("Reversed String is : ");
	fmt.Printf("%s",arr);
}